### ISSUE TITLE
simplify code with dplyr

### ISSUE THREAD

**Issue by soodoku** [dcomtois/summarytools#7]
Hey,

Great package!

I think code for descr etc. can be radically simplified using dplyr.

For instance:
```
iris <- read_csv("https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data", col_names = F)
iris_num <- iris %>%
   summarize_if(is.numeric, funs(mean = mean, median = median, min = min, max = max, missing = sum(is.na(.)))) 
iris_num_long <- iris_num %>%
  gather(key = "key", value = "words") %>%
  separate(key, into = c("var", "statistic")) %>%
  spread(key = "var", value = "words")

```

produces

```
 iris_num_long
# A tibble: 5 x 5
  statistic    X1    X2    X3    X4
* <chr>     <dbl> <dbl> <dbl> <dbl>
1 max        7.90  4.40  6.90 2.50 
2 mean       5.84  3.05  3.76 1.20 
3 median     5.80  3.00  4.35 1.30 
4 min        4.30  2.00  1.00 0.100
5 missing    0     0     0    0    
```

and this allows you to pass arbitrary functions to summarize easily

**Comment by dcomtois** [comment]
Hi Gaurav and thanks for your interest in summarytools! I'm taking a note for maybe using dplyr in the future, if I find time for it. Up to now I stuck mostly with base functions (and I have reservations about pipes for various reasons).

**Comment by dcomtois** [comment]
I spend some time reorganizing most of the code so as to make it more manageable / understandable, and I also used dplyr in both `descr()` and `freq()`, which gives an edge on performance. I also used `dplyr::n_distinct()` in `dfSummary()` as it's way faster than `sum(duplicates())`. It's in the dev-current branch so anyone who has a chance to test it out, that would be very helpful.

Just install the dev-current version this way:

`devtools::install_github("dcomtois/summarytools", ref = "dev-current")`

Thanks

**Comment by dcomtois** [comment]
I've updated the dev-current branch. It should fix the issue. Thanks for testing and reporting, really appreciated!

**Comment by dcomtois** [comment]
Bonjour Cécile,

Okay I'll need to check out why this happens... Just to validate, are these indeed IFT-14 codes?

For the graph sizes, they are too small so the value for graph.magnif should be >1, probably between 1.25 and 1.50 in your case.

Merci :)

**Comment by cecilesauder** [comment]
Hello !
Well, according to the definition found on wikipedia
<https://en.wikipedia.org/wiki/ITF-14>, no, there aren't IFT-14 codes.
I've tried to change the graph.magnif value to 1.5 but it doesn't change at
all the graph sizes :(

Merci à toi !

Le lun. 17 déc. 2018 à 17:45, Dominic Comtois <notifications@github.com> a
écrit :

**Comment by dcomtois** [comment]
yeah I realized that after posting my comment (which I edited). For the graph.magnif, it has to be an argument to print, not dfSummary:

print(dfSummary(data, style = 'grid', plain.ascii = FALSE), graph.magnif = 1.5,, method = 'render', headings = FALSE)

**Comment by dcomtois** [comment]
Yeah I realized after posting my last comment, which I edited. And ignore my last deleted comment... your syntax is correct. I'm really not sure why you don't see changes in the graphs. Maybe try another yaml config, like this one:

```
output: 
  rmarkdown::html_vignette: 
    css: 
    - !expr system.file("rmarkdown/templates/html_vignette/resources/vignette.css", package = "rmarkdown")
    - !expr system.file("includes/stylesheets/summarytools.css", package = "summarytools")
```

Let me know if that helps!

**Comment by dcomtois** [comment]
Actually try a value of 3 or 4, that could be the answer after all.

**Comment by cecilesauder** [comment]
I add the css files in the yaml => no changes
I put the value to 4 ==> no changes

I note that it's only the rendering results which doesn't take into account
this graph.magnif value, because in the preview of Rstudio, graphs are
enormous :D

Le mar. 18 déc. 2018 à 10:35, Dominic Comtois <notifications@github.com> a
écrit :

**Comment by dcomtois** [comment]
Hehe ok... If you expand your window where the markdown has been generated into html, you'll see the size of the graphs adjust... but I need to figure out how to force those cells to keep their size.

In the meantime, there's always the non-rendering method (, but then you loose the histograms...

**Comment by cecilesauder** [comment]
If we could choose the size of the column it would be very great, because
my variable column is too large too :D
I love the tiny histograms, there are so cute 😍

Le mar. 18 déc. 2018 à 11:47, Dominic Comtois <notifications@github.com> a
écrit :

**Comment by dcomtois** [comment]
@cecilesauder See issue #57 :)
